core/mm/mobj.c

```diff
@@ -133,6 +134,40 @@ static struct mobj_phys *to_mobj_phys(struct mobj *mobj)
 	return container_of(mobj, struct mobj_phys, mobj);
 }
 
+#ifdef CFG_CORE_RWDATA_NOEXEC
+static struct mobj *mobj_phys_alloc_tee_ram(paddr_t pa, size_t size,
+					    uint32_t cattr,
+					    enum teecore_memtypes area_type)
+{
+	void *va = NULL;
+	struct mobj_phys *moph = NULL;
+
+	if ((area_type != MEM_AREA_TEE_RAM_RX) &&
+			(area_type != MEM_AREA_TEE_RAM_RW)) {
+		DMSG("cannot allocate specified area_type");
+		return NULL;
+	}
+
+	va = phys_to_virt(pa, area_type);
```

Comment:
This is copy/paste from `mobj_phys_alloc()`. Please use common code for this, either by extending `mobj_phys_alloc()` as needed or by factoring out a helper function. I believe the former should be more useful.